lib/kaguya/channel.ex

```diff
@@ -127,6 +127,11 @@ defmodule Kaguya.Channel do
     {:stop, :normal, :ok, nil}
   end
 
+  def handle_call(:join, _from, {name, _users, _buffer} = state) do
+    Util.joinChan(name)
+    {:reply, :ok, state}
```

Comment:
I think we should clear the user and buffer state here since there's no real guarantee of what's happened since a reconnect.